read-fonts/src/tables/postscript/blend.rs

```diff
@@ -0,0 +1,166 @@
+//! Support for the dictionary and charstring blend operator.
+
+use font_types::{BigEndian, F2Dot14, Fixed};
+
+use super::Error;
+use crate::tables::variations::{ItemVariationData, ItemVariationStore, VariationRegionList};
+
+/// The maximum number of region scalars that we precompute.
+const MAX_PRECOMPUTED_SCALARS: usize = 16;
+
+/// State for processing the blend operator for DICTs and charstrings.
+///
+/// To avoid allocation, scalars are computed on demand but this can be
+/// prohibitively expensive in charstrings where blends are applied to
+/// large numbers of elements. To amortize the cost, a fixed number of
+/// precomputed scalars are stored internally and the overflow is computed
+/// as needed.
+///
+/// The `MAX_PRECOMPUTED_SCALARS` constant determines the size of the
+/// internal buffer (currently 16).
+pub struct BlendState<'a> {
+    store: ItemVariationStore<'a>,
+    regions: VariationRegionList<'a>,
+    coords: &'a [F2Dot14],
+    vs_index: u16,
+    needs_update: bool,
```

Comment:
Not a huge fan of this. Needs a comment to explain why it's a good idea.

I don't know the usage pattern but I'd prefer we avoided it unless that is somehow infeasible. Could we:

1. Have two types - or an enum with two variants - one for the unupdated/stale state and one for the updated state. Provide methods to move between them.
   * Makes sense if it's desirable to change vs_index repeatedly and only call update once. If you more or less call update_if_needed every change seems silly.
1. Have one type that is always updated
   * Makes sense if the usage pattern is to essentially always call update_if_needed when you set a vs index

Reply:
Fair enough. I changed this to compute scalars on construction and update them directly when the store index changes.